test/framework/src/test/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertionsTests.java

```diff
@@ -52,4 +59,161 @@ public void writeTo(StreamOutput out) throws IOException {
             throw new IllegalArgumentException("Not supported.");
         }
     }
+
+    public void testAssertXContentEquivalent() throws IOException {
+        XContentBuilder original = JsonXContent.contentBuilder();
+        original.startObject();
+        addRandomNumberOfFields(original);
+        {
+            original.startObject(randomAlphaOfLength(10));
+            addRandomNumberOfFields(original);
+            original.endObject();
+        }
+        {
+            original.startArray(randomAlphaOfLength(10));
+            int elements = randomInt(5);
+            for (int i = 0; i < elements; i++) {
+                original.value(randomAlphaOfLength(10));
+            }
+            original.endArray();
+        }
+        original.endObject();
+
+        XContentParser parser = XContentHelper.createParser(NamedXContentRegistry.EMPTY, original.bytes(), original.contentType());
+        XContentBuilder copy = JsonXContent.contentBuilder();
+        parser.nextToken();
+        XContentHelper.copyCurrentStructure(copy.generator(), parser);
+        assertToXContentEquivalent(original.bytes(), copy.bytes(), original.contentType());
```

Comment:
Makes sense, I added shuffling the copy, assertToXContentEquivalent should still work.